android/src/main/java/com/amplitude/reactnative/AmplitudeReactNativeModule.java

```diff
@@ -335,6 +336,23 @@ public void setPlan(String instanceName, ReadableMap planProperties, Promise pro
             promise.resolve(true);
         }
     }
+    
+    @ReactMethod
+    public void setIngestionMetadata(String instanceName, ReadableMap ingestionMetadataProperties, Promise promise) {
+        IngestionMetadata ingestionMetadata = new IngestionMetadata();
+        if (properties.hasKey("sourceName")) {
+            ingestionMetadata.setBranch(ingestionMetadataProperties.getString("sourceName"));
```

Comment:
setBranch?

Reply:
Thanks for reviewing! Good catch, updated!